Electronic Colloquium on Computational Complexity.js

```diff
@@ -1,94 +1,122 @@
 {
 	"translatorID": "09a9599e-c20e-a405-d10d-35ad4130a426",
 	"label": "Electronic Colloquium on Computational Complexity",
-	"creator": "Jonas Schrieb",
-	"target": "^https?://eccc\\.weizmann\\.ac\\.il/",
+	"creator": "Jonas Schrieb and Morgan Shirley",
+	"target": "^https?://eccc\\.weizmann\\.ac\\.il/(title|year|keyword|report)",
 	"minVersion": "1.0.0b3.r1",
 	"maxVersion": "",
 	"priority": 100,
 	"inRepository": true,
 	"translatorType": 4,
-	"browserSupport": "gcsib",
-	"lastUpdated": "2017-01-05 17:11:41"
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2023-05-31 20:54:48"
 }
 
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2023 Jonas Schrieb and Morgan Shirley
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
 function detectWeb(doc, url) {
-	var singleRe   = /^https?:\/\/eccc\.weizmann\.ac\.il\/report\/\d{4}\/\d{3}/;
 	var multipleRe = /^https?:\/\/eccc\.weizmann\.ac\.il\/(title|year|keyword)\//;
-	if (singleRe.test(url)) {
-		return "report";
-	} else if (multipleRe.test(url)) {
+	if (multipleRe.test(url)) {
 		return "multiple";
 	}
+	else if (ZU.fieldIsValidForType('title', 'preprint')) {
+		return "preprint";
+	}
+	else {
+		return "report";
+	}
+}
+
+// Get a meta tag from the header
+function metaXPath(name) {
+	return `/html/head/meta[contains(@name,'citation_${name}')]/@content`;
```

Comment:
I understand what you're trying to do, but I don't think this is a great approach here.
a) Using a CSS selector with `attr` is barely longer than calling your custom function and more readable and
b) if the translator uses a lot of `citation_` "highwire" metatags, I'd recommend calling the embedded metadata translator and letting that do most of the work. There's a template for this availabel in the translator editor/Scaffold

Reply:
Switched to using the Embedded Metadata translator. The template code in Scaffold does not work. Specifically, the getTranslatorObject function appears to fail silently. I fixed this by using the getTranslatorObject code from the OSF Preprint translator mentioned above.